### PR TITLE
coreutils: switch to version 9 for darwin

### PR DESCRIPTION
**Pull request by happysalada**
###### Motivation for this change

switch darwin to version 9 of coreutils.
Currently testing this.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by happysalada** [comment]
@dasJ might be interested. When I tested your original PR it was including this, but somehow this was reverted later. This PR is just to bring it back.
@r-burns thank you for bringing my attention to this, I had completely missed it.

**Comment by happysalada** [comment]
Ok just finished testing on darwin-x86_64. Leaving this for one more day open, just in case people want to give feedback.

**Comment by mohe2015** [comment]
I thought we wanted to do this in the next cycle as the Darwin workers on hydra are broken again.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -4239,13 +4239,7 @@ with pkgs;
 
   cpcfs = callPackage ../tools/filesystems/cpcfs { };
 
-  # coreutils 9 messes up some binary format on darwin
-  # https://github.com/NixOS/nixpkgs/pull/141684
-  # test by trying to build alacritty for example
```
Comment (risicle) [inline pkgs/top-level/all-packages.nix:4244]: Anyone happen to have an example which doesn't have quite as massive dependency/rebuild footprint for testing?

Comment (happysalada) [reply pkgs/top-level/all-packages.nix:4244]: r-burns was saying `jx` was the fastest one for him.
My machine is quite old, it still took me 6 hours I think.
If you leave it overnight, it should be fine.
I made sure by also testing `alacritty`.

**Comment by happysalada** [comment]
@mohe2015 I'm happy to do it in the next cycle. it would be a shame to miss the update I felt.
I'm happy to go with what people feel is best.

**Comment by mohe2015** [comment]
Others please also comment.

I'm not a good hydra reader but it seems like only one x86 Darwin builder is working and two are hanging so I assume the reasoning is still valid.

I don't know though how long it would take or so.

**Comment by SuperSandro2000** [comment]
Was the original issue why this was reverted fixed? If so how?

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (r-burns) [reply pkgs/top-level/all-packages.nix:4244]: No, my jx testing was all on linux. I didn't encounter any issues on darwin.

**Comment by r-burns** [comment]
As far as I'm aware, @happysalada was the only one to reproduce the darwin errors locally, so if those are fixed then this is good to go IMO. But if we want to wait until the next staging cycle, that's fine by me.

**Comment by happysalada** [comment]
@SuperSandro2000 the original issue was that the resulting rust binaries had the wrong binary format and couldn't be executed. With the coreutils fix, this doesn't happen anymore (this is what was tested in this PR).

**Comment by SuperSandro2000** [comment]
You mean with https://github.com/NixOS/nixpkgs/commit/924ccbff80dbfba51b9b893879062fcf2539cd72#diff-88e1a963530d2a05ed02f38aa91b0a292d425396118e1d1a7b05e9132d4f7f5e? Or which fix exactly?

**Comment by happysalada** [comment]
@SuperSandro2000 yes, that fix.

**Comment by happysalada** [comment]
I'm going to say let's merge this one, the build queue is almost empty, and it removes weird logic that is unecessary.

**Pull request merged**